test/runner.js

```diff
@@ -0,0 +1,192 @@
+'use strict'
+
+let fs = require('fs')
+let path = require('path')
+let expect = require('chai').expect
+let rimraf = require('rimraf')
+let Launcher = require('webdriverio/build/lib/launcher')
+let parseXmlString = require('xml2js').parseString
+
+let configFile = './test/fixtures/wdio.conf.js'
+let resultsDir = path.join(__dirname, '../allure-results') 
+
+describe('"before all" hook', () => {
+
+  beforeEach(() => {
+    clean()
+  })
+
+  it('should not show up in the results when it is passing', () => {
+
+    return run(['before-all-passing']).then((code) => {
+
+      return getResultsXML()
+        .then((results) => {
+
+          expect(results).to.have.lengthOf(1)
+          expect(results[0]['ns2:test-suite']['test-cases']).to.have.lengthOf(1)
+          
+          results[0]['ns2:test-suite']['test-cases'][0]['test-case']
+            .forEach((testCase) => {
+
+              switch(testCase.name[0]) {
+
+                case '"before" hook':
+                case '"before all" hook':
+                case '"before each" hook':
+                case '"after all" hook':
+                case '"after each" hook':
+                  throw new Error('\'' + testCase.name + '\' was not expected in the results when no hook fails')
+                  break
+
+              }
+
+            })
+
+        })
+      
+    })
+
+  })
+
+  it('should marked broken when failing', () => {
+
+    return run(['before-all-failing']).then((code) => {
+
+      return getResultsXML()
+        .then((results) => {
+
+          expect(results).to.have.lengthOf(1)
+          expect(results[0]['ns2:test-suite']['test-cases']).to.have.lengthOf(1)
+          
+          let isBeforeAllHookBroken = false
+
+          results[0]['ns2:test-suite']['test-cases'][0]['test-case']
+            .forEach((testCase) => {
+
+              switch(testCase.name[0]) {
+
+                case '"before" hook':
+                case '"before each" hook':
+                case '"after all" hook':
+                case '"after each" hook':
+                  throw new Error('\'' + testCase.name + '\' was not expected in the results when no hook fails')
+                  break
+
+                case '"before all" hook':
```

Comment:
Good thought. It is not necessary to follow `chaijs` API you can just create a helper function `findHook`

```js
const beforeAllHook = findHook('beforeEach')
expect(beforeAllHook).to.have.deep.property('$.status', 'broken')

expect(findHook('before all')).to.be.undefined
expect(findHook('before')).to.be.undefined
```